src/pages/FlagCommentPage.js

```diff
@@ -123,14 +117,30 @@ function FlagCommentPage(props) {
         },
     ];
 
+    const getActionToFlag = useCallback(() => {
+        let reportAction = props.reportActions[`${props.route.params.reportActionID.toString()}`];
+
+        // Handle threads if needed
+        if (reportAction === undefined || reportAction.reportActionID === undefined) {
+            reportAction = ReportActionsUtils.getParentReportAction(props.report);
+        }
+
+        return reportAction;
+    }, [props.report, props.reportActions, props.route.params.reportActionID]);
+
     const flagComment = (severity) => {
         let reportID = getReportID(props.route);
+        const reportAction = getActionToFlag();
 
         // Handle threads if needed
-        if (reportAction === undefined || reportAction.reportActionID === undefined) {
+        if (ReportUtils.isChatThread(props.report)) {
```

Comment:
I'm not sure this will work right - we only want the report to be the parent report IF the reportAction being flagged from the thread is the parent action. This seems like it will always use the parent report for ANY flags from a thread. Unless I'm misunderstanding?

Reply:
Hmm I think you're right. So it should be something like the block below, is that right?

```
if (ReportUtils.isChatThread(props.report) && reportAction.reportActionID === ReportActionsUtils.getParentReportAction(props.report).reportActionID) {
    reportID = ReportUtils.getParentReport(props.report).reportID;
}
```

Reply:
That seems right to me!